src/main/java/org/zendesk/client/v2/Zendesk.java

```diff
@@ -1749,7 +1753,82 @@ public void deleteOrganizationMembership(long user_id, long organization_members
 
         return new PagedIterable<>(templateUri, handleList(type, "results"));
     }
-
+    
+    /**
+     * Search API implementation with pagination support. 
+     * 
+     * @param String  query string used filter a type given by searchType
+     * @param Map<String, Object> additional parameters other than filter string like per_page, page etc
+     * @param String name of any field of the searchType
+     * @param SortOrder
+     * @param Class<?> type of search entity like Ticket, User etc
+     * @param Class<T> page return type to which the search result will be deserialized 
+     */
+    public <T> Optional<T> getSearchPageResults(
+        @NotNull final String query,
+        @Nullable final Map<String, Object> queryParams,
+        @Nullable final String sortBy,
+        @Nullable final SortOrder sortOrder,
+        @NotNull final Class<?> searchType,
+        @NotNull final Class<T> pageType
+        ) {
+
+      final Map<String, Object> paramsMap = new HashMap<>();
+      
+      if(queryParams != null) {
+        paramsMap.putAll(queryParams);
+      }
+      
+      if(sortBy!=null && sortOrder!=null) {
+        paramsMap.put("sort_by", sortBy);
+        paramsMap.put("sort_order", sortOrder.getQueryParameter());
+      }
+
+      
+      final StringBuilder uriTemplate = new StringBuilder("/search.json{?query");
+
+      for (final String paramName : paramsMap.keySet()) {
+          uriTemplate.append(",")
+                  .append(paramName);
+      }
+
+      uriTemplate.append("}");
+      
+      String typeName = null;
```

Comment:
Actually, this loop could also be extracted and shared with `getSearchResults`, since the only difference is `searchType.equals(entry.getValue())` vs `type.equals(entry.getValue())` and that var could be a parameter of the extracted private method